src/components/inputs/money-input/money-input.js

```diff
@@ -175,27 +199,29 @@ export const createMoneyValue = (currencyCode, amount) => {
   };
 };
 
-const formatAmount = (amount, currencyCode) => {
+const getAmountAsNumberFromMoneyValue = money =>
+  money.type === 'highPrecision'
+    ? money.preciseAmount / 10 ** money.fractionDigits
+    : money.centAmount / 10 ** currencies[money.currencyCode].fractionDigits;
+
+// gets called with a string and should return a formatted string
+const formatAmount = (value, currencyCode, locale) => {
   // fallback in case the user didn't enter an amount yet (or it's invalid)
-  const moneyValue = createMoneyValue(currencyCode, amount) || {
+  const moneyValue = createMoneyValue(currencyCode, value) || {
```

Comment:
Should we have an `locale` invariant here too?

Reply:
`formatAmount` is currently only called from places where either
- `locale` is already checked
- the instance method where the locale is available as prop-type checks warn if not

So right now the `invariant` prevents mistakes from `ui-kit` consumers, but we don't get warnings in case it's used wrong inside this file itself. Not sure if it's worth adding a check. I don't have a strong opinion.

Maybe those checks should ALL be moved to `formatAmount`? Then we'd always run them in either case. I'll move them 👍

Reply:
On second thought, this would spread the checks across different functions which is less nice. I think I like the current version better.